lib/browserid.js

```diff
@@ -3,12 +3,24 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 
+const Joi = require('joi');
+
 const AppError = require('./error');
 const config = require('./config');
 const logger = require('./logging')('assertion');
 const P = require('./promise');
 
 const HEX_STRING = /^[0-9a-f]+$/;
+const CLAIMS_SCHEMA = Joi.object({
+  'uid': Joi.string().length(32).regex(HEX_STRING).required(),
+  'fxa-generation': Joi.number().integer().min(0).required(),
+  'fxa-verifiedEmail': Joi.string().max(255).required(),
+  'fxa-lastAuthAt': Joi.number().integer().min(0).required(),
+  'fxa-tokenVerified': Joi.boolean().optional(),
+  'fxa-amr': Joi.array().items(Joi.string().alphanum()).optional(),
```

Comment:
Should be a little more strict and allow only `email, pwd, otp` here?

Reply:
I'm in two minds about this.  It seems worthwhile for sanity-checking, but the oauth-server doesn't actually interpret this values at all, it just passes them on to the reliers.  If we ever add another method, do we want to require a coordinated oauth-server deploy, or just get it "for free" when the auth-server updates?  I lean towards the latter.

Reply:
> oauth-server doesn't actually interpret this values at all

Consider me convinced!